### PR TITLE
Fixes #29536: Use puma-systemd-plugin to provide systemd notify support

### PR DESCRIPTION
**Pull request by ehelms**
This points to a fork of puma-systemd-plugin maintained by theforeman
organization to ensure this works with Puma 4.

### CODE REVIEW

**Comment by ehelms** [comment]
Packaging change -- https://github.com/theforeman/foreman-packaging/pull/5718

**Comment by ehelms** [comment]
Looks like bundler is making this strategy painful and forcing me towards releasing the gem under a different name. Unless anybody has any ideas? Note this gem is only used for production.

 @ekohl for your awareness

File: config/puma/production.rb [hunk 1]
```diff
@@ -13,6 +13,10 @@
 #
 workers ENV.fetch('FOREMAN_PUMA_WORKERS', 2).to_i
 
+# Provides systemd notify support through the
+# puma-systemd-plugin
+plugin :systemd
```
Comment (ekohl) [inline config/puma/production.rb]: Does this raise an exception if it can't be loaded?

**Review by ekohl — COMMENTED** [review]
I'd agree with a forked name, but should we give the maintainer a few days to respond?

**Comment by ehelms** [comment]
I'll ask again on the PRs as we are sitting at 7 days.

File: config/puma/production.rb (hunk 1 above)
Comment (ehelms) [reply config/puma/production.rb]: Aye, I am catching that now.

**Review by ekohl — APPROVED** [review]
:+1: pending packaging.

**Comment by ekohl** [comment]
https://github.com/sj26/puma-plugin-systemd#watchdog is that also something we want to use?

**Comment by ehelms** [comment]
Given we have got some traction from the maintainer of the plugin I have switched this to be as if that plugin was used directly with a new release. There are currently 1 needed and 1 optional PRs in discussion right now:

 * (Required) https://github.com/sj26/puma-plugin-systemd/pull/11
 * (Optional) https://github.com/sj26/puma-plugin-systemd/pull/12

**Comment by ehelms** [comment]
If I read WatchdogSec support correctly that enables systemd to fail the service if it does not respond to a ping. In which case, I would think we'd want to have `Restart=on-watchdog` added to restart if and only if watchdog is the thing that kills the process.

**Comment by ekohl** [comment]
I have no experience with using systemd's watchdog so I'm ok with leaving it as a separate issue, but now that we have the support, it's good to at least consider it.

**Comment by ehelms** [comment]
Packaging is ready to be merged and built -- https://github.com/theforeman/foreman-packaging/pull/5718
Followed by this PR being merged
Lastly, updating Foreman dependency -- https://github.com/theforeman/foreman-packaging/pull/5728

**Comment by ehelms** [comment]
[test katello]

**Comment by ehelms** [comment]
Katello failure is unrelated

**Pull request merged**